addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js

```diff
@@ -448,6 +427,109 @@ QUnit.test('chatter should become enabled when creation done', async function (a
     );
 });
 
+QUnit.test('Form view not scrolled when switching record', async function (assert) {
+    assert.expect(6);
+
+    this.data['res.partner'].records = [{
+        activity_ids: [],
+        id: 1,
+        display_name: "Partner 1",
+        description: [...Array(60).keys()].join('\n'),
+        message_ids: [],
+        message_follower_ids: [],
+    }, {
+        activity_ids: [],
+        id: 2,
+        display_name: "Partner 2",
+        message_ids: [],
+        message_follower_ids: [],
+    }];
+
+    const messages = [...Array(60).keys()].map(id => {
+        return {
+            author_id: [10, "Demo User"],
+            body: `<p>Message ${id + 1}</p>`,
+            date: "2019-04-20 10:00:00",
+            id: id + 1,
+            message_type: 'comment',
+            model: 'res.partner',
+            record_name: `Partner ${id % 2 ? 1 : 2}`,
+            res_id: id % 2 ? 1 : 2,
+        };
+    });
+    this.data['mail.message'].records = messages;
+
+    await this.createView({
+        data: this.data,
+        hasView: true,
+        // View params
+        View: FormView,
+        model: 'res.partner',
+        arch: `
+            <form string="Partners">
+                <sheet>
+                    <field name="name"/>
+                    <field name="description"/>
+                </sheet>
+                <div class="oe_chatter">
+                    <field name="message_ids"/>
+                </div>
+            </form>
+        `,
+        viewOptions: {
+            currentId: 1,
+            ids: [1, 2],
+        },
+        config: {
+            device: { size_class: config.device.SIZES.LG },
+        },
+        env: {
+            device: { size_class: config.device.SIZES.LG },
+        },
+    });
+
+    const controllerContentEl = document.querySelector('.o_content');
+    const formViewEl = document.querySelector('.o_form_view');
+
+    assert.strictEqual(
+        document.querySelector('.breadcrumb-item.active').textContent,
+        'Partner 1',
+        "Form view should display partner 'Partner 1'"
+    );
+    assert.strictEqual(controllerContentEl.scrollTop, 0,
+        "The top of the form view is visible"
+    );
+    await afterNextRender(async () => {
+        controllerContentEl.scrollTop = controllerContentEl.scrollHeight - controllerContentEl.offsetHeight;
+        await triggerEvent(
+            document.querySelector('.o_ThreadViewer_messageList'),
+            'scroll'
+        );
+    });
```

Comment:
To improve readability of tests, we put no newlines between successive interactions then followed by successive assertions, and we put newline between last assertion and following interaction:

```js
await start();
assert1();
assert2();

await doSomething1();
await doSomething2();
assert3();
assert4();

await doSomething3();
assert5();
```